samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -197,20 +208,27 @@ static void producer_thread(void)
 
 		zbus_chan_notify(&bm_channel, K_MSEC(200));
 	}
-	uint32_t duration = (k_uptime_get_32() - start);
+
+#if defined(CONFIG_ARCH_POSIX)
+	uint32_t end_ns = native_rtc_gettime_us(RTC_CLOCK_PSEUDOHOSTREALTIME) * NSEC_PER_USEC;
+#else
+	uint32_t end_ns = k_cyc_to_ns_near32(sys_clock_cycle_get_32());
+#endif
+
+	uint32_t duration = end_ns - start_ns;
```

Comment:
Same as the `start_ns` case.

Reply:
Same as above, defined a single `GET_ARCH_TIME_NS()` and called it in both instances.